Core/views.py

```diff
@@ -87,20 +88,33 @@ class RegistrationView(FormView):
     success_url = reverse_lazy('home')
 
     form_class = RegistrationForm
-
+    
     event = Event.objects.filter(active=True).first()
+    # event=''
 
     def post(self, request, *args, **kwargs):
         form = RegistrationForm(request.POST)
+        print(((dict(request.POST))['student_number'])[0])
         if form.is_valid():
-            form.save()
-            messages.add_message(request, messages.SUCCESS,
+            try:
+                Registration.objects.get(student_number=((dict(request.POST))['student_number'])[0], event=self.event)
+                print("1")
+                alert='Already Registered!!'
+                return render(request, 'registration.html', {'form': form, 'event': self.event, 'alert':alert})
+            except: 
+                form.save()
+                print("2")
+                messages.add_message(request, messages.SUCCESS,
                                  "Successfully registered.")
-            return redirect(reverse_lazy('home'))
-        return render(request, 'registration.html', {'form': form, 'event': self.event})
+                return redirect(reverse_lazy('registration'))
+                # return render(request, 'registration.html', {'form': form, 'event': self.event, 'messages':messages})
+        else:
+            # form=self.form_class()
```

Comment:
Like @apoorvaeternity said remove comment that were used when debugging.
I don't know if the print("1")  and print("2") are really needed, but I would try to avoid them as much as possible.